lib/ts/fetch.ts

```diff
@@ -224,7 +242,7 @@ export default class AuthHttpRequest {
                 doNotDoInterception = !shouldDoInterceptionBasedOnUrl(
                     WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getOrigin(),
                     AuthHttpRequest.config.apiDomain,
```

Comment:
check if the auth header added by the user is the same as the one we have - and if it is, then we remove the user added header and just let the interceptors take care of adding it